services/importer/lib/importer/connector.rb

```
@@ -108,27 +108,55 @@ def self.check_availability!(user)
 
       private
 
-      ACCEPTED_PARAMETERS = %w(connection schema table sql_query sql_count encoding).freeze
-      SERVER_OPTIONS = %w(dsn driver host port database).freeze
+      ACCEPTED_PARAMETERS = %w(connection schema table sql_query sql_count encoding columns).freeze
       REQUIRED_PARAMETERS = %w(connection table).freeze
 
+      SERVER_OPTIONS = %w(dsn driver host server address port database).freeze
+      USER_OPTIONS   = %w(uid pwd user username password).freeze
+
+      def fetch_ignoring_case(hash, key)
```

just asking, the simplest thing to do is to let it raise an exception and even do not check for "nilness" as it is supposed to be a hash always.